kafka-connect-aws-s3/src/test/scala/io/lenses/streamreactor/connect/aws/s3/formats/BytesFormatWriterTest.scala

```diff
@@ -99,97 +89,14 @@ class BytesFormatWriterTest extends AnyFlatSpec with Matchers {
-      MessageDetail(byteArrayValue,
-                    ByteArraySinkData("notUsed".getBytes, None),
-                    Map.empty,
-                    Some(Instant.now()),
-                    topic,
-                    0,
-                    Offset(0),
-      ),
-    )
-
-    outputStream.toByteArray should be(bytes)
-
-    bytesFormatWriter.complete()
-
-  }
-
-  "convert" should "write binary with KeyAndValueWithSizes" in {
-
-    val outputStream      = new S3ByteArrayOutputStream()
-    val bytesFormatWriter = new BytesFormatWriter(outputStream, BytesWriteMode.KeyAndValueWithSizes)
-    bytesFormatWriter.write(MessageDetail(byteArrayValue,
-                                          byteArrayValue,
-                                          Map.empty,
-                                          Some(Instant.now()),
-                                          topic,
-                                          0,
-                                          Offset(0),
-    ))
-
-    outputStream.toByteArray should be(pixelLengthBytes ++ pixelLengthBytes ++ bytes ++ bytes)
-
-    bytesFormatWriter.complete()
-
-  }
-
-  "convert" should "write binary with KeyWithSize" in {
-
-    val outputStream      = new S3ByteArrayOutputStream()
-    val bytesFormatWriter = new BytesFormatWriter(outputStream, BytesWriteMode.KeyWithSize)
-
-    bytesFormatWriter.write(
-      MessageDetail(byteArrayValue,
-                    ByteArraySinkData("notUsed".getBytes, None),
-                    Map.empty,
-                    Some(Instant.now()),
-                    topic,
-                    0,
-                    Offset(0),
-      ),
-    )
-
-    outputStream.toByteArray should be(pixelLengthBytes ++ bytes)
-
-    bytesFormatWriter.complete()
-
-  }
-
-  "convert" should "write binary with ValueWithSize" in {
-
-    val outputStream      = new S3ByteArrayOutputStream()
-    val bytesFormatWriter = new BytesFormatWriter(outputStream, BytesWriteMode.ValueWithSize)
-    bytesFormatWriter.write(
-      MessageDetail(ByteArraySinkData("notUsed".getBytes, None),
-                    byteArrayValue,
-                    Map.empty,
-                    Some(Instant.now()),
-                    topic,
-                    0,
-                    Offset(0),
-      ),
-    )
-
-    outputStream.toByteArray should be(pixelLengthBytes ++ bytes)
-
-    bytesFormatWriter.complete()
-
-  }
-
-  "convert" should "write  multiple parts of an image and combine" in {
+  "convert" should "write multiple parts of an image and combine" in {
 
     val stream = classOf[BytesFormatWriter].getResourceAsStream("/streamreactor-logo.png")
     val bytes: Array[Byte] = IOUtils.toByteArray(stream)
     val (bytes1, bytes2) = bytes.splitAt(bytes.length / 2)
 
     val outputStream      = new S3ByteArrayOutputStream()
-    val bytesFormatWriter = new BytesFormatWriter(outputStream, BytesWriteMode.ValueOnly)
+    val bytesFormatWriter = new BytesFormatWriter(outputStream)
```

Comment:
This writer should not allow writing a second record, it's another guard.

Reply:
done.